### PR TITLE
[Executor] Refine _exec in batch engine

### PR DESCRIPTION
**Pull request by Jasmin3q**
- Seperate previous_line_results and line_results as input and output for clarity.
- Move line_results and aggr_results to the end of the param list and add default value.
- Avoid unnecessary log when not in a resume run.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/batch/_batch_engine.py [hunk 1]
```diff
@@ -315,9 +315,9 @@ async def _exec_in_task(
 
     async def _exec(
```
Comment (thy09) [inline src/promptflow/promptflow/batch/_batch_engine.py:316]: We can reorder the parameters like this:
```
batch_inputs,
run_id,
output_dir,
raise_on_line_failure,
line_results = None,
aggr_results = None,
```

**Review by guming-learning — DISMISSED** [review]

File: src/promptflow/promptflow/batch/_batch_engine.py [hunk 2]
```diff
@@ -322,12 +329,13 @@ async def _exec_in_task(
...
         self,
-        line_results: List[LineResult],
-        aggr_result: AggregationResult,
         batch_inputs: List[Dict[str, Any]],
         run_id: str = None,
         output_dir: Path = None,
         raise_on_line_failure: bool = False,
+        previous_line_results: List[LineResult] = None,
+        line_results: List[LineResult] = [],  # serve as output
+        aggr_result: AggregationResult = AggregationResult({}, {}, {}),  # serve as output
```
Comment (thy09) [inline src/promptflow/promptflow/batch/_batch_engine.py:338]: `serve as output` is not that clear, add more comments

**Review by thy09 — DISMISSED** [review]

File: src/promptflow/promptflow/batch/_batch_engine.py [hunk 3]
```diff
@@ -322,12 +329,14 @@ async def _exec_in_task(
 
     async def _exec(
         self,
-        line_results: List[LineResult],
-        aggr_result: AggregationResult,
         batch_inputs: List[Dict[str, Any]],
         run_id: str = None,
         output_dir: Path = None,
         raise_on_line_failure: bool = False,
+        previous_line_results: List[LineResult] = None,  # Input parameter: previous line results for resume run
```
Comment (PeiwenGaoMS) [inline src/promptflow/promptflow/batch/_batch_engine.py:336]: Extract a standard function comment？

Comment (Jasmin3q) [reply src/promptflow/promptflow/batch/_batch_engine.py]: Added docstring for _exec.

**Review by PeiwenGaoMS — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**